File: easy/problems/0219-contains-duplicate-ii/solution.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
    const seen = new Set<number>();
    for (let i = 0; i < nums.length; i++) {
        if (seen.has(nums[i])) {
            return true;
        }
        seen.add(nums[i]);
        if (seen.size > k) {
            seen.delete(nums[i - k]);
        }
    }
    return false;
}

File: easy/problems/0219-contains-duplicate-ii/solution.py
def containsNearbyDuplicate(nums: list[int], k: int) -> bool:
    seen = set()
    for i in range(len(nums)):
        if nums[i] in seen:
            return True
        seen.add(nums[i])
        if len(seen) > k:
            seen.remove(nums[i - k])
    return False
